Continue Cleanup of install tests

permcody opened the pull request:
1) Removing old --copy-tests and --tests options
2) Getting rid of convoluted logic to run tests in system installed location
3) Introducing the first C++17 filesystem usage
4) Remove the first C++17 filesystem usage 😢 

refs #19022

permcody committed:
Continue Cleanup of install tests

1) Removing old --copy-tests and --tests options
2) Getting rid of convoluted logic to fun tests in system installed location
3) Introducing the first C++17 filesystem syntax

refs #19022

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/988651/) on 4dade4a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20595/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/988659/) on 4dade4a wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>83ee3d</th>
      <th colspan=4>#20595 4dade4</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.51%</td>
      <td>82.52%</td>
      <td>+0.01%</td>
      <td>64.29%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73571</td>
      <td>73569</td>
      <td>-2</td>
      <td>18</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15594</td>
      <td>15588</td>
      <td>-6</td>
      <td>10</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20595/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20595/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20595/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 64.29% is less than the suggested 90.0%



This comment will be updated on new commits.

permcody commented:
The coverage report is a little bit depressing. I did remove a bunch of untested lines, which is why the coverage actually increases despite the relatively low new line coverage...

permcody committed:
Dropping first C++17 include :(

refs #19022

permcody commented on framework/src/base/MooseApp.C:
```diff
@@ -1468,53 +1459,48 @@ MooseApp::copyInputs()
+    std::filesystem::path working_dir;
+    try
+    {
+      working_dir = std::filesystem::current_path();
```
@fdkong - This is what I just pushed up and it worked fine on the Mac.

permcody committed:
INSTALLABLE_DIRS default fix

INSTALLABLE_DIRS is not defined in any .mk file. It does not need to be
stored on the Makefile stack. Also the test needs to account fo undefined
variables, not just empty variables.

refs #19022
